### DISCUSSION TITLE
Why do NaN and Inf appear in the simulation? I have already made modifications according to the suggested methods.

### DISCUSSION THREAD

**Discussion by niubiworker** [NanoComp/meep#2934] [Q&A]
I am running a simulation based on the example of a bent waveguide, but I don't know why the simulation is showing errors. Is there an issue with the way I'm using it?

```
import meep as mp
import matplotlib.pyplot as plt

pml=3
resolution=20
field_decay=1e-3
lambda_max=1.5
lambda_min=1.6
fcen = 0.5*(1/lambda_max+1/lambda_min)
df = 1/lambda_min-1/lambda_max
str_length=8
str_width=1.58
h=0.3
nfreq=101
LN = mp.Medium(epsilon_diag=(2.2111**2,2.2111**2,2.1376**2))

sim_group=[39.28830000000001, 0.3, 8.09779893013697]
eig_sources=[mp.EigenModeSource(
    src=mp.GaussianSource(fcen,fwidth=df),
    center=mp.Vector3(0.5*str_length,0,0),
    size=mp.Vector3(0,5*h,2*str_width),
    eig_parity=mp.EVEN_Y+mp.ODD_Z,
    eig_match_freq=True,
    direction=mp.X,
    )]

symmetry=[mp.Mirror(mp.Z,phase=-1),mp.Mirror(mp.Y,phase=1)]
monitor1_loc=mp.Vector3(x=0.5*str_length+0.5)
monitor2_loc=mp.Vector3(x=sim_group[0]-0.5*str_length)
monitor_area=mp.Vector3(0,7*h,3*str_width)
sim_area=mp.Vector3(sim_group[0]+2*pml,10*sim_group[1]+2*pml,sim_group[2]+2*pml)
sim_center=mp.Vector3(sim_group[0]*0.5)
geo_str=[mp.Block(size=mp.Vector3(sim_group[0],h,str_width),
                         center=mp.Vector3(0.5*sim_group[0],0,0),
                         material=LN)]
sim = mp.Simulation(cell_size=sim_area,
                    geometry_center=sim_center,
                    resolution=resolution,
                    boundary_layers=[mp.PML(thickness=pml)],
                    sources=eig_sources,
                    geometry=geo_str,
                    symmetries=symmetry,
                    )

refl_str=sim.add_flux(fcen,df,nfreq,
                mp.FluxRegion(center=monitor1_loc,direction=mp.X,size=monitor_area))
tran_str=sim.add_flux(fcen,df,nfreq,
                mp.FluxRegion(center=monitor2_loc,direction=mp.X,size=monitor_area))
pt=monitor2_loc-mp.Vector3(x=0.5)

sim.plot2D(output_plane=mp.Volume(center=mp.Vector3(x=0.5*sim_group[0]),
                                  size=mp.Vector3(sim_group[0]+2*pml,0,sim_group[2]+2*pml)))
plt.show()
sim.plot2D(output_plane=mp.Volume(center=mp.Vector3(x=0.5*sim_group[0]),
                                  size=mp.Vector3(sim_group[0]+2*pml,10*sim_group[1]+2*pml,0)))
plt.show()
sim.plot2D(output_plane=mp.Volume(center=mp.Vector3(x=0.5*str_length),
                                  size=mp.Vector3(0,10*sim_group[1]+2*pml,sim_group[2]+2*pml)))
plt.show()

sim.run(until_after_sources=mp.stop_when_fields_decayed(50,mp.Ey,pt,field_decay))
```

'Warning: grid volume is not an integer number of pixels; cell size will be rounded to nearest pixel.
Using MPI version 4.1, 32 processes
     block, center = (19.6442,0,0)
          size (39.2883,0.3,1.58)
          axes (1,0,0), (0,1,0), (0,0,1)
          dielectric constant epsilon diagonal = (4.88896,4.88896,4.56933)
Warning: grid volume is not an integer number of pixels; cell size will be rounded to nearest pixel.
     block, center = (19.6442,0,0)
          size (39.2883,0.3,1.58)
          axes (1,0,0), (0,1,0), (0,0,1)
          dielectric constant epsilon diagonal = (4.88896,4.88896,4.56933)
Warning: grid volume is not an integer number of pixels; cell size will be rounded to nearest pixel.
     block, center = (19.6442,0,0)
          size (39.2883,0.3,1.58)
          axes (1,0,0), (0,1,0), (0,0,1)
          dielectric constant epsilon diagonal = (4.88896,4.88896,4.56933)

Initializing structure...
Halving computational cell along direction y
Halving computational cell along direction z
Splitting into 32 chunks by voxels
time for choose_chunkdivision = 0.00113486 s
Working in 3D dimensions.
Computational cell is 45.3 x 9 x 14.1 with resolution 20
     block, center = (19.6442,0,0)
          size (39.2883,0.3,1.58)
          axes (1,0,0), (0,1,0), (0,0,1)
          dielectric constant epsilon diagonal = (4.88896,4.88896,4.56933)
Warning: grid volume is not an integer number of pixels; cell size will be rounded to nearest pixel.
time for set_epsilon = 0.595986 s

Warning: grid volume is not an integer number of pixels; cell size will be rounded to nearest pixel.
MPB solved for frequency_1(1.41164,0,0) = 0.8182 after 20 iters
MPB solved for frequency_1(1.02971,0,0) = 0.644855 after 8 iters
MPB solved for frequency_1(1.03184,0,0) = 0.645833 after 5 iters
MPB solved for frequency_1(1.03184,0,0) = 0.645833 after 1 iters
on time step 1 (time=0.025), 23.2909 s/step
on time step 49 (time=1.225), 0.0843644 s/step
on time step 97 (time=2.425), 0.0842138 s/step
on time step 145 (time=3.625), 0.0841403 s/step
on time step 192 (time=4.8), 0.0852058 s/step
on time step 240 (time=6), 0.0848948 s/step
on time step 288 (time=7.2), 0.0849695 s/step
on time step 336 (time=8.4), 0.0843097 s/step
meep: simulation fields are NaN or Inf
Abort(1) on node 18 (rank 18 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 18'

Thank you very much!

**Comment by niubiworker** [comment]
Although I tried according to the recommendations from the official Meep documentation, the issue remains unresolved. Additionally, I noticed that simulations proceed smoothly when only Gaussian sources are used. However, when eigenmode sources are used, the simulation results in NaN. Why is this happening?
Of course, I also tried without symmetry, but the issue still couldn't be resolved.

**Reply by smartalecH** [reply]
When you removed the simulation symmetries, did you also remove the eigenmode source parity conditions? If not, try that too. 

Based on your statement, it sounds like the eigenmode source is having problems. (I think you were trying to say that a uniform plane current source with a Gaussian time profile works fine). 

Maybe try plotting the mode itself. 

Might be worth iterating in 2D first too.

**Comment by niubiworker** [comment]
I find that make `src=mp.GaussianSource(fcen,fwidth=df)` to `src=mp.GaussianSource(fcen,fwidth=0.2*fcen)`,it is become true.
